src/attributes.jl

```diff
@@ -0,0 +1,280 @@
+
+"""
+Main structure for holding attributes in e.g. plots!
+"""
+struct ObservableAttributes
+    # name, for better error messages!
+    name::String
+    # We dont have one node per value anymore, but instead one node
+    # that gets triggered on any setindex!, or whenever an input attribute node changes
+    # This makes it easier to layer Observable{ObservableAttributes}()
+    on_change::Observable{Pair{Symbol, Any}}
+    # The supported fields, so we can throw an error, whenever fields are not supported
+    supported_fields::Set{Symbol}
+    # The attributes given at construction time, taking the highest priority
+    from_user::Dict{Symbol, Any}
+    # attributes filled in by e.g. a theme, or other processes in the pipeline
+    from_theme::Dict{Symbol, Any}
+    # The "global" default values for this specific attribute instance
+    # Will be immutabe (maybe not a Dict then?) and shared between all similar objects
+    default_values::Dict{Symbol, Any}
```

Comment:
```suggestion
"""
    ObservableAttributes(; attrs...)
Main structure for holding attributes in e.g. plots!

## Extended help

### Fields
$(FIELDS)
"""
struct ObservableAttributes
    "name, for better error messages!"
    name::String
    """
    We dont have one node per value anymore, but instead one node
    that gets triggered on any setindex!, or whenever an input attribute node changes
    This makes it easier to layer Observable{ObservableAttributes}()
    """
    on_change::Observable{Pair{Symbol, Any}}
    "The supported fields, so we can throw an error, whenever fields are not supported"
    supported_fields::Set{Symbol}
    "The attributes given at construction time, taking the highest priority"
    from_user::Dict{Symbol, Any}
    "attributes filled in by e.g. a theme, or other processes in the pipeline"
    from_theme::Dict{Symbol, Any}
    """
    The "global" default values for this specific attribute instance
    Will be immutable (maybe not a Dict then?) and shared between all similar objects
    """
    default_values::Dict{Symbol, Any}
```

This integrates DocStringExtensions, so that you can see what the type is in the documentation and REPL.